python/rikai/io.py

```diff
@@ -49,6 +49,39 @@ def _normalize_uri(uri: str) -> str:
     ).geturl()
 
 
+def _gcsfs():
```

Comment:
Use `@functools.cache` instead of maintaining the `_GCSFS` global variable?

Reply:
TIL

Reply:
`@functools.cache` is new from python 3.9...
We need to provide compatibility from 3.7 tho.

https://docs.python.org/3/library/functools.html

Other alternatives?

Reply:
guess we can use `lru`  🤫